xlator/xlator.go

```diff
@@ -0,0 +1,150 @@
+package xlator
+
+/*
+#cgo LDFLAGS: -ldl
+
+#include <stdlib.h>    // free()
+#include <dlfcn.h>     // dlopen(), dlclose()
+#include "options.h"   // volume_option_t
+*/
+import "C"
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"unsafe"
+)
+
+const (
+	maxXlatorOptions = 100
```

Comment:
`afr.so` with 49 options.

`maxXlatorOptions` is used to create a pointer to a slice of that size. It's not allocated in memory.